doc/website/blog/2022-03-14-ent-hooks-examples.md

```diff
@@ -0,0 +1,430 @@
+			NotEmpty(),
+		field.String("password").
+			Sensitive().
+			NotEmpty(),
+	}
+}
+
+// Edges of the User.
+func (User) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.To("pets", Dog.Type),
+		edge.To("cache", Cache.Type).Unique(),
+	}
+}
+```
+```go
+// Dog holds the schema definition for the Dog entity.
+type Dog struct {
+	ent.Schema
+}
+
+// Fields of the Dog.
+func (Dog) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("name").
+			NotEmpty(),
+		field.Int("owner_id").
+			Optional(),
+	}
+}
+
+// Edges of the Dog.
+func (Dog) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.From("owner", User.Type).
+			Ref("pets").
+			Field("owner_id").
+			Unique(),
+	}
+}
+```
+```go
+// Cache holds the schema definition for the Cache entity.
+type Cache struct {
+	ent.Schema
+}
+
+// Fields of the Cache.
+func (Cache) Fields() []ent.Field {
+	return []ent.Field{
+		field.Int("walks"),
+	}
+}
+```
+
+For each example I'll provide a test and a recipe, I highly suggest starting with a test to see that the hook does what you need it to do.
+Unlike many other parts of the codebase, hooks do one thing only and should be simple to test.
+
+### Keeping our secrets safe:
+
+Many times, we need to store strings that include secrets, a good example can be a connection
+string to a database, usually in the form of ```mysql://root:pass@localhost:3306)```. Here, for example, it’s very
+easy to see that we don't want the password to unintentionally be retrieved by a query unless required by the server
+(for example to connect to a database).
+The expected behavior is best described by this test:
+```go
+func TestUserConnectionStringHook(t *testing.T) {
+	ctx := context.Background()
+	c := enttest.Open(t, dialect.SQLite,
+		"file:TestSchemaConfHooks?mode=memory&cache=shared&_fk=1",
+	)
+	u := c.User.Create().SetName("Yoni").SetConnectionString("mysql://root:pass@localhost:3306)").SaveX(ctx)
+	require.Equal(t, "mysql://root:****@localhost:3306)", u.ConnectionString)
+	require.Equal(t, "pass", u.Password)
+	require.Equal(t, "mysql://root:pass@localhost:3306)", u.FullConnectionString())
+}
+```
+
+As you can see, inserting a connection string updates the string to have `****` instead of the password and stores the 
+password in the ```Password``` field that is defined as [sensitive](https://entgo.io/docs/schema-fields#sensitive-fields).
+To have the connection string in full valid form we created a utility function ```FullConnectionString()```.
+
+#### Recipe:
+
+First thing, add the hook to your schema:
+```go
+// Hooks of the User.
+func (User) Hooks() []ent.Hook {
+	return []ent.Hook{
+		hook.If(clearConnectionString,
+			hook.And(hook.Or(hook.HasOp(ent.OpUpdateOne), hook.HasOp(ent.OpCreate)), hook.HasFields(user.FieldConnectionString)),
+		),
+	}
+}
+```
+* Do not forget to run go generate or the hook will not be registered.
+
+A few hook helpers are used here, ```clearConnectionString``` is the name of the hook function, and it will run if a user 
```

Comment:
dont need triple backticks for a "variable", one is enough..

Reply:
👍